### DISCUSSION TITLE
Variable to return the current workflow filename

### DISCUSSION THREAD

**Discussion by kenorb** [actions/runner#3250] [Q&A]
Is there some variable available which can return the name of the current workflow file (e.g. `.github/workflows/test.yml`)?

**Comment by ShreeChaturvedi** [comment]
Use `job.workflow_file_path`. It is the path from the repo root, like `.github/workflows/test.yml`.

```yaml
- run: echo "${{ job.workflow_file_path }}"
```

`github.workflow` is the workflow name. If the file has no `name:`, you get the path instead.

For the full string with branch, use `github.workflow_ref` (example: `owner/repo/.github/workflows/file.yml@refs/heads/branch`) and parse the path out.

Not available on GitHub Enterprise Server.

Docs: [Contexts reference](https://docs.github.com/en/actions/reference/workflows-and-actions/contexts)
